src/pexprs-toArgumentNameList.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+// --------------------------------------------------------------------
+// Imports
+// --------------------------------------------------------------------
+
+var common = require('./common');
+var pexprs = require('./pexprs');
+
+// --------------------------------------------------------------------
+// Operations
+// --------------------------------------------------------------------
+
+/*
+  The `toArgumentNameList` method returns a list of strings that will
+  be used as the default argument names for its receiver (a pexpr) in a
+  semantic action. This is used exclusively by the Semantics Editor.
+*/
+
+pexprs.PExpr.prototype.toArgumentNameList = common.abstract;
+
+pexprs.Prim.prototype.toArgumentNameList = function(optIndex) {
```

Comment:
Is the argument here really optional? It doesn't seem to be, since it's used in the `else` branch below.